internal/generated/api/retry_http_client.go

```diff
@@ -0,0 +1,331 @@
+//
+// 4. All Other HTTP Status Codes:
+//   - Returns immediately without retry attempts
+//   - Preserves the original response and any errors
+//
+// Context Cancellation:
+// The method respects context cancellation during backoff delays. If the request
+// context is cancelled or times out during a delay, the method returns the last
+// response along with the context error.
+//
+// Parameters:
+//   - req: The HTTP request to execute. Must not be nil.
+//
+// Returns:
+//   - *http.Response: The HTTP response from the server (may be from original or retry attempt)
+//   - error: Any error that occurred during the request or retry process
+//
+// Thread Safety:
+// This method is safe for concurrent use by multiple goroutines.
+func (t *RetryTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Buffer the body so we can retry non-idempotent requests as well.
+	var bodyBytes []byte
+	if req.Body != nil {
+		b, err := io.ReadAll(req.Body)
+		if err != nil {
+			return nil, err
+		}
+		if closeErr := req.Body.Close(); closeErr != nil {
+			return nil, closeErr
+		}
+		bodyBytes = b
+	}
+
+	makeReq := func(ctx context.Context) *http.Request {
+		r := req.Clone(ctx)
+		if bodyBytes != nil {
+			r.Body = io.NopCloser(bytes.NewReader(bodyBytes))
+		}
+		return r
+	}
+
+	retryCount := 0
+
+	for {
+		r := makeReq(req.Context())
+		res, err := t.base().RoundTrip(r)
+		if err != nil {
+			return nil, err
+		}
+
+		switch res.StatusCode {
+		case http.StatusTooManyRequests:
+			if retryCount >= maxRetryAttempts {
+				return res, nil
+			}
+
+			var backoff time.Duration
+			// Respect Retry-After header if present, otherwise use exponential backoff.
+			if ra := res.Header.Get("Retry-After"); ra != "" {
+				if secs, err := strconv.Atoi(ra); err == nil && secs > 0 {
+					backoff = time.Second * time.Duration(secs)
+				} else {
+					backoff = calculateBackoff(retryCount)
+				}
+			} else {
+				backoff = calculateBackoff(retryCount)
+			}
+
+			// Drain and retry.
+			if _, err := io.Copy(io.Discard, res.Body); err != nil {
+				return res, err
+			}
+			if err := res.Body.Close(); err != nil {
+				return res, err
+			}
+			if err := sleepOrDone(req.Context(), backoff); err != nil {
+				return res, err
+			}
+			retryCount++
+			continue
+		case http.StatusGatewayTimeout:
+			if retryCount >= maxRetryAttempts {
+				return res, nil
+			}
+			// Peek body to check for special code 7001 (do not retry).
+			b, err := io.ReadAll(res.Body)
+			if err != nil {
+				return res, err
+			}
+			if err := res.Body.Close(); err != nil {
+				return res, err
+			}
+			// Restore body for caller if we return.
+			res.Body = io.NopCloser(bytes.NewReader(b))
+
+			var apiErr struct {
+				Code int `json:"code"`
+			}
+			if err := json.Unmarshal(b, &apiErr); err != nil {
+				// If we can't parse the error, treat it as a regular 504 and retry.
+				apiErr.Code = 0
+			}
+			if apiErr.Code == 7001 {
+				return res, nil
+			}
```

Comment:
error should be returned here otherwise it causes regression.  see previous client implementation:

https://github.com/couchbasecloud/terraform-provider-couchbase-capella/blob/main/internal/resources/gsi.go#L238